bears/xml/XMLBear.py

```diff
@@ -15,8 +36,19 @@ def process_output(self, output, filename, file):
         if self.stderr_output:  # pragma: no cover
             return self._process_issues(self.stderr_output, filename)
 
-    def run(self, filename, file):
+    def run(self, filename, file,
+            schema: path="",
+            dtdvalid: path_or_url(path, str)=""):
         '''
-        Checks the code with `xmllint`.
+        Checks the code with ``xmllint``.
+
+        :param: schema:        Use a W3C XML Schema file named ``schema`` for
+                               validation.
+        :param: dtdvalid:      Use the ``DTD`` specified by URL for validation.
         '''
+        if schema:
```

Comment:
You need to put `arguments = "{filename}"` on top here.
otherwise the things you add using `+=` will be saved to the class (as arguments is a class variable) and will be used next time an object is created from it